### PR TITLE
#14873: test for groupby.agg coercing booleans

### LINKED ISSUE #14873: BUG: groupby.agg coercing booleans
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

dat = pd.DataFrame({'a': [1, 1, 2, 2], 'b': [0, 1, 2, 3], 'c': [None, None, 1, 1]})

gp = dat.groupby('a')

case_one = gp['b'].aggregate(lambda x: (x != 0).all())

case_two = gp['c'].aggregate(lambda x: x.isnull().all())

```
#### Problem description
By comparing ``case_one`` and ``case_two``, it seems that ``aggregate`` function turns boolean values sometimes to integers 1 or 0.

For ``case_one`` we get,
```
a
1    False
2     True
Name: b, dtype: bool
```
However for ``case_two`` we get
```
a
1    1.0
2    0.0
Name: c, dtype: float64
```
#### Expected Output
For ``case_two`` we should get

```
a
1    True
2     False
Name: b, dtype: bool
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-53-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_AU.UTF-8
LOCALE: en_AU.UTF-8

pandas: 0.19.1
nose: 1.3.7
pip: 8.1.2
setuptools: 23.0.0
Cython: None
numpy: 1.11.0
scipy: 0.17.0
statsmodels: None
xarray: None
IPython: 4.2.0
sphinx: None
patsy: None
dateutil: 2.5.3
pytz: 2016.4
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None

</details>

This same issue appears in a different environment as well.

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-53-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_AU.UTF-8

pandas: 0.18.1
nose: 1.3.7
pip: 9.0.1
setuptools: 23.0.0
Cython: None
numpy: 1.11.2
scipy: 0.18.1
statsmodels: None
xarray: None
IPython: 5.0.0
sphinx: None
patsy: None
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.1.3
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by TrigonaMinima**
- [x] closes #14873 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

The referenced issue seems to be solved in the current master. So added a test for the edge case.

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -75,6 +75,20 @@ def test_basic(dtype):
...
+    case_one_res = gp['b'].aggregate(lambda x: (x != 0).all())
+    case_two_res = gp['c'].aggregate(lambda x: x.isnull().all())
+
+    index = Index([1, 2], name='a')
+    case_one_exp = Series([False, True], index=index, name='b')
+    case_two_exp = Series([True, False], index=index, name='c')
+
+    assert_series_equal(case_one_res, case_one_exp)
+    assert_series_equal(case_two_res, case_two_exp)
+
```
Comment (gfyoung) [inline pandas/tests/groupby/test_groupby.py]: Can we just make this a separate test?

In general, we are trying to move toward smaller, more modular tests.

Comment (TrigonaMinima) [reply pandas/tests/groupby/test_groupby.py]: @gfyoung created a separate test.

**Review by gfyoung — APPROVED** [review]
cc @jreback

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -1698,3 +1698,19 @@ def test_groupby_agg_ohlc_non_first():
     result = df.groupby(pd.Grouper(freq='D')).agg(['sum', 'ohlc'])
 
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_agg_coercing_bools():
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py]: can you move this to pandas/tests/groupby/aggregate/test_agg.py

File: pandas/tests/groupby/test_groupby.py [hunk 3]
```diff
@@ -1698,3 +1698,19 @@ def test_groupby_agg_ohlc_non_first():
...
+    # issue 14873
+    dat = pd.DataFrame(
+        {'a': [1, 1, 2, 2], 'b': [0, 1, 2, 3], 'c': [None, None, 1, 1]})
+    gp = dat.groupby('a')
+    case_one_res = gp['b'].aggregate(lambda x: (x != 0).all())
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py]: can you write this a 2 successve cases, using
result = 
expected = 
assert_series_equal(result, expected)

for each one (can be in the same test)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_groupby.py (hunk 3 above)
Comment (TrigonaMinima) [reply pandas/tests/groupby/test_groupby.py]: @jreback made the suggested changes.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @TrigonaMinima

**Pull request merged**